playwright/navigation.spec.ts

```diff
@@ -0,0 +1,21 @@
+import { expect, test } from '@playwright/test';
+
+test('Navigate and check content', async ({ page }) => {
+  await page.goto('/');
+  await expect(page.getByTestId('intro')).toBeVisible();
+  await expect(page.getByTestId('projects')).toBeVisible();
+  await expect(page.getByTestId('about')).toBeVisible();
+  await expect(page.getByTestId('contact')).toBeVisible();
+
+  await page.getByRole('link', { name: 'About' }).click();
+  await page.waitForURL('/#about');
+  await expect(page.getByTestId('about')).toBeInViewport();
```

Comment:
The `.toBeInViewport()` assertion checks whether the element with `the data-test-id` attribute is visible in the current viewport.